### ISSUE TITLE
Tranfserwise: api v1 end-of-life August 31st

### ISSUE THREAD

**Issue by ayushin** [OCA/bank-statement-import#402]
Looks like the current module will stop working August 31st...

```
Your Wise API integration needs to be upgraded to stop using v1/transfers (funding) OR v1/statements (retrieval) so that it doesn’t stop working on August 31st 2021. 

In September 2020 we notified you that v1/transfers (funding) and v1/statements endpoints were being deprecated in favour of new API versions that support Strong Customer Authentication (SCA) checks v3/transfers and v3/statements. 

In April 2021 there was additional communication reminding customers of the importance of upgrading their Wise API Integration to comply with SCA checks. 

If you continue to use these APIs after August 31st your Wise API Integration will stop working which could prevent funding transfers or reconciling statements. 

You can reach out to api@wise.com if you have any questions about using our V3 APIs. 

Thanks, 
Joe Armstrong
Product Manager for Wise Platform
```

**Comment by alexey-pelykh** [comment]
Yes, this module needs an upgrade

**Comment by ayushin** [comment]
is anybody working on this or am I the only one using it?

**Comment by alexey-pelykh** [comment]
The upgrade has been scheduled, but it still works for us so not an immediate priority for maintenance. If someone steps-up with sponsoring the prioritization, obviously that would happen sooner.

**Comment by ayushin** [comment]
i can't help with the sponsoring, but perhaps with testing

**Comment by alexey-pelykh** [comment]
thanks! I'll update the issue once there's a PR addressing the subject

**Comment by llabusch93** [comment]
I think the priority is now urgent, as the module does not work anymore:

```
production_1  | 2021-09-01 14:08:06,924 15 INFO production odoo.addons.account_bank_statement_import_online.models.online_bank_statement_provider: Scheduled pull of online bank statements... 
production_1  | 2021-09-01 14:08:06,929 15 INFO production odoo.addons.account_bank_statement_import_online.models.online_bank_statement_provider: Pulling online bank statements of: TransferWise, TransferWise 
production_1  | 2021-09-01 14:08:07,102 15 WARNING production odoo.addons.account_bank_statement_import_online.models.online_bank_statement_provider: Online Bank Statement Provider "TransferWise.com" failed to obtain statement data since 2021-08-01 00:00:00 until 2021-09-01 00:00:00 
production_1  | Traceback (most recent call last):
production_1  |   File "/addons/global_subs/bank-statement-import/account_bank_statement_import_online/models/online_bank_statement_provider.py", line 182, in _pull
production_1  |     statement_date_until
production_1  |   File "/addons/global_subs/bank-statement-import/account_bank_statement_import_online_transferwise/models/online_bank_statement_provider_transferwise.py", line 105, in _obtain_statement_data
production_1  |     data = self._transferwise_retrieve(url, api_key)
production_1  |   File "/addons/global_subs/bank-statement-import/account_bank_statement_import_online_transferwise/models/online_bank_statement_provider_transferwise.py", line 272, in _transferwise_retrieve
production_1  |     with self._transferwise_urlopen(url, api_key) as response:
production_1  |   File "/addons/global_subs/bank-statement-import/account_bank_statement_import_online_transferwise/models/online_bank_statement_provider_transferwise.py", line 287, in _transferwise_urlopen
production_1  |     return urllib.request.urlopen(request)
production_1  |   File "/usr/lib/python3.5/urllib/request.py", line 163, in urlopen
production_1  |     return opener.open(url, data, timeout)
production_1  |   File "/usr/lib/python3.5/urllib/request.py", line 472, in open
production_1  |     response = meth(req, response)
production_1  |   File "/usr/lib/python3.5/urllib/request.py", line 582, in http_response
production_1  |     'http', request, response, code, msg, hdrs)
production_1  |   File "/usr/lib/python3.5/urllib/request.py", line 510, in error
production_1  |     return self._call_chain(*args)
production_1  |   File "/usr/lib/python3.5/urllib/request.py", line 444, in _call_chain
production_1  |     result = func(*args)
production_1  |   File "/usr/lib/python3.5/urllib/request.py", line 590, in http_error_default
production_1  |     raise HTTPError(req.full_url, code, msg, hdrs, fp)
production_1  | urllib.error.HTTPError: HTTP Error 410: Gone
production_1  | 2021-09-01 14:08:07,315 15 WARNING production odoo.addons.account_bank_statement_import_online.models.online_bank_statement_provider: Online Bank Statement Provider "TransferWise.com" failed to obtain statement data since 2021-08-01 00:00:00 until 2021-09-01 00:00:00 
production_1  | Traceback (most recent call last):
production_1  |   File "/addons/global_subs/bank-statement-import/account_bank_statement_import_online/models/online_bank_statement_provider.py", line 182, in _pull
production_1  |     statement_date_until
production_1  |   File "/addons/global_subs/bank-statement-import/account_bank_statement_import_online_transferwise/models/online_bank_statement_provider_transferwise.py", line 105, in _obtain_statement_data
production_1  |     data = self._transferwise_retrieve(url, api_key)
production_1  |   File "/addons/global_subs/bank-statement-import/account_bank_statement_import_online_transferwise/models/online_bank_statement_provider_transferwise.py", line 272, in _transferwise_retrieve
production_1  |     with self._transferwise_urlopen(url, api_key) as response:
production_1  |   File "/addons/global_subs/bank-statement-import/account_bank_statement_import_online_transferwise/models/online_bank_statement_provider_transferwise.py", line 287, in _transferwise_urlopen
production_1  |     return urllib.request.urlopen(request)
production_1  |   File "/usr/lib/python3.5/urllib/request.py", line 163, in urlopen
production_1  |     return opener.open(url, data, timeout)
production_1  |   File "/usr/lib/python3.5/urllib/request.py", line 472, in open
production_1  |     response = meth(req, response)
production_1  |   File "/usr/lib/python3.5/urllib/request.py", line 582, in http_response
production_1  |     'http', request, response, code, msg, hdrs)
production_1  |   File "/usr/lib/python3.5/urllib/request.py", line 510, in error
production_1  |     return self._call_chain(*args)
production_1  |   File "/usr/lib/python3.5/urllib/request.py", line 444, in _call_chain
production_1  |     result = func(*args)
production_1  |   File "/usr/lib/python3.5/urllib/request.py", line 590, in http_error_default
production_1  |     raise HTTPError(req.full_url, code, msg, hdrs, fp)
production_1  | urllib.error.HTTPError: HTTP Error 410: Gone
production_1  | 2021-09-01 14:08:07,328 15 INFO production odoo.addons.account_bank_statement_import_online.models.online_bank_statement_provider: Scheduled pull of online bank statements complete.
```

**Comment by ayushin** [comment]
We should consider https://github.com/OCA/bank-statement-import/issues/412 instead

**Comment by alexey-pelykh** [comment]
Addressed via #413 and #414
